File: frontend/src/models/Flight.ts
import { Aircraft } from './Aircraft';
import { Seat } from './Seat';

export interface Flight {
    id: number;
    flightNumber: string;
    origin: string;
    destination: string;
    departureTime: string;
    arrivalTime: string;
    price: number;
    aircraft?: Aircraft;
    seats?: Seat[];
}

export interface FlightSearchParams {
    destination?: string;
    date?: string;
    maxPrice?: number;
}

File: src/main/java/com/cgi/flightbooking/models/Flight.java
package com.cgi.flightbooking.models;


import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
public class Flight {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String flightNumber;
    private String origin;
    private String destination;
    private LocalDateTime departureTime;
    private LocalDateTime arrivalTime;
    private BigDecimal price;

    // Many flights can use the same aircraft model
    @ManyToOne
    private Aircraft aircraft;

    // One flight has many seats
    @OneToMany(mappedBy = "flight", cascade = CascadeType.ALL)
    private List<Seat> seats;
}
